js/lib/index.js

```diff
@@ -0,0 +1,703 @@
+   *   "IPI": "International Parties Information", // may be null
+   *   "ISNI": "International Standard Name Identifier", // may be null
+   *   "street_address": "street address for payments", // may be null
+   * }
+   * ```
+   *
+   * Only previously set objects using the same private key as this instance may
+   * be overwritten.
+   *
+   * @param {object} individual - the individual identity to be set
+   * @param {string} individual.name - the natural key of the individual
+   * @param {string} [individual.IPI] - optional International Parties
+   * Information
+   * @param {string} [individual.ISNI] - optional International Standard Name
+   * Identifier
+   * @param {string} [individual.street_address] optional street address for
+   * payments
+   *
+   * @returns {Promise<BatchStatus>} a promise for the result of transaction submission
+   *
+   * @since 1.0.0
+   */
+  setIndividual (individual) {
+    return _submitOmiTransaction(this._sawtoothRestUrl,
+                                 this._privateKey,
+                                 'SetIndividualIdentity',
+                                 IndividualIdentity,
+                                 'name',
+                                 individual)
+  }
+
+  /**
+   * Fetch an IndividualIdentity from the global state, and returns the value
+   * via a promise.
+   *
+   * The resulting object returned on the resolved promise will be structured
+   * like so:
+   *
+   * ```
+   * {
+   *   "name": "String", // a Unique name; the natural key
+   *   "IPI": "International Parties Information", // may be null
+   *   "ISNI": "International Standard Name Identifier", // may be null
+   *   "street_address": "street address for payments", // may be null
+   *   "pubkey": "creators pub key in hex" // matches the signer's pubkey
+   * }
+   * ```
+   *
+   * @param {string} name - the individual's name (the natural key)
+   *
+   * @returns {Promise<Object>} a promise for an IndividualIdentity
+   *
+   * @since 1.0.0
+   */
+  getIndividual (name) {
+    return _omiStateEntry(this._sawtoothRestUrl, IndividualIdentity, name)
+  }
+
+  /**
+   * Returns a cursor for all the users.
+   */
+  getIndividuals () {
+    return _omiStateCursor(this._sawtoothRestUrl, IndividualIdentity)
+  }
+
+  /**
+   * Set Organizational Identity.
+   *
+   * A SetOrganizationalIdentity transaction will be submitted to the
+   * blockchain, and a URL for the resulting batch status will be returned via a
+   * promise.
+   *
+   * The argument should be structured like so:
+   *
+   * ```
+   * {
+   *   "name": "String", // a Unique name; the natural key
+   *   "type": <OrganizationalIdentity.type>, // the type of the organization
+   *   "IPI": "International Parties Information", // may be null
+   *   "street_address": "street address for payments", // may be null
+   * }
+   * ```
+   *
+   * Only previously set objects using the same private key as this instance may
```

Comment:
Probably should specify at the outset of this docstring that it can both create a new Organization and update an existing one.